src/core/abstract.ml

```diff
@@ -10,10 +10,10 @@ let build_abstract a = match a.a_impl with
 	| None -> ()
 
 let has_direct_to uctx ab pl b =
-	List.exists (unify_to {uctx with allow_transitive_cast = false} ab pl b) ab.a_to
+	List.exists (unify_to {uctx with allow_implicit_cast = false} ab pl b) ab.a_to
```

Comment:
Shouldn't this be `{uctx with allow_transitive_cast = false; allow_implicit_cast = false;}`?

Reply:
If false `allow_transitive_cast ` is passed to `unify_to` [then](https://github.com/HaxeFoundation/haxe/blob/526bd66a61b9f93af55643132cc4e7424b78308d/src/core/tUnification.ml#L891) an equality check is used, not a unification one. This will make passing `allow_implicit_cast` meaningless. With false `allow_implicit_cast` - there still be unification check, just one that does not allow abstract casts at all, so there will be no transitive abstract casts.